astester.imports.js

```diff
@@ -30,31 +66,94 @@ export function local(memory) {
     massa: {
       memory,
       assembly_script_generate_event(string) {
-        console.log("event", getString(string));
+        console.log('event', getString(string));
+      },
+
+      assembly_script_set_data(k_ptr, v_ptr) {
+        const k = getString(k_ptr);
+        const v = getString(v_ptr);
+        const addressStorage = ledger.get(defaultAddress).storage;
+        addressStorage.set(k, v);
       },
 
       assembly_script_set_data_for(a_ptr, k_ptr, v_ptr) {
         const a = getString(a_ptr);
         const k = getString(k_ptr);
         const v = getString(v_ptr);
-        if (!Storage.has(a)) {
-          Storage.set(a, new Map());
+        if (!ledger.has(a)) {
+          ledger.set(a, emptyAddress);
         }
-        const addressStorage = Storage.get(a);
+
+        const addressStorage = ledger.get(a).storage;
         addressStorage.set(k, v);
       },
+
+      assembly_script_get_data(k_ptr) {
+        let v = '';
+        const k = getString(k_ptr);
+        if (ledger.has(defaultAddress)) {
+          const addressStorage = ledger.get(defaultAddress).storage;
+          if (addressStorage.has(k)) {
+            v = addressStorage.get(k);
+          }
+        }
+        return newString(v);
+      },
+
       assembly_script_get_data_for(a_ptr, k_ptr) {
-        let v = "";
+        let v = '';
         const a = getString(a_ptr);
         const k = getString(k_ptr);
-        if (Storage.has(a)) {
-          const addressStorage = Storage.get(a);
+        if (ledger.has(a)) {
+          const addressStorage = ledger.get(a).storage;
           if (addressStorage.has(k)) {
             v = addressStorage.get(k);
           }
         }
         return newString(v);
       },
+
+      assembly_script_has_data(k_ptr) {
+        const k = getString(k_ptr);
+        const addressStorage = ledger.get(defaultAddress).storage;
+        return addressStorage.has(k);
+      },
+
+      assembly_script_has_data_for(a_ptr, k_ptr) {
+        const a = getString(a_ptr);
+        const k = getString(k_ptr);
+        if (ledger.has(a)) {
+          const addressStorage = ledger.get(a).storage;
+          return addressStorage.has(k);
+        } else {
+          return false;
+        }
+      },
+
+      assembly_script_get_call_stack() {
+        const [_, contractAddress] = ledger.keys();
+        return newString(
+          '[ ' + defaultAddress + ' , ' + contractAddress + ' ]',
+        );
+      },
+
+      assembly_script_unsafe_random() {
+        return BigInt(
+          Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(),
+        );
+      },
+
+      // map the assemblyscript file with the contractAddresses generated
+      assembly_script_create_sc(path_ptr) {
```

Comment:
What do you mean by the SC content ? 
The .wasm or the as file content ?

Reply:
This mocking function is adding a new entry to the datastore.
I'm wondering why you do that.

Reply:
Following our discussion, do you think you still need to do that ?

Reply:
As for now no, 
I had idead but maybe not for now.
It would be nice to be able to pass a callback function in parameter no ? WDYT ? 
We can maybe create an issue for that

Reply:
Yeah, create an issue please